src/opds_dumper.cpp

```diff
@@ -287,10 +287,13 @@ std::string OPDSDumper::noJSDownloadLinks(const std::string& bookId) const
   );
 }
 
-std::string OPDSDumper::noJSData() const
+std::string OPDSDumper::noJSData(kiwix::Filter filter) const
```

Comment:
The API of dumping in plain HTML format diverges from the API of dumping in OPDS format. Why not stick to the signature of the `dumpOPDSFeed2()` and obtain the book ids in `InternalServer` (probably using `InternalServer::search_catalog()`)?

Reply:
`InternalServer::search_catalog()` doesnt allow me to change the filter :<< I need to clear the category